src/suites/cts/validation/createView.spec.ts

```diff
@@ -0,0 +1,268 @@
+  { arrayLayerCount: 0, baseArrayLayer: ARRAY_LAYER_COUNT_2D, success: false },
+  // It is an error for the array layer range of the view to exceed that of the texture
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D + 1, baseArrayLayer: 0, success: false },
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, baseArrayLayer: 1, success: false },
+  { arrayLayerCount: 2, baseArrayLayer: ARRAY_LAYER_COUNT_2D - 1, success: false },
+  { arrayLayerCount: 1, baseArrayLayer: ARRAY_LAYER_COUNT_2D, success: false },
+]);
+
+g.test(
+  'default values validates the same as specifying values for more than 1 array layer',
+  async t => {
+    const { format, dimension, arrayLayerCount, mipLevelCount, success } = t.params;
+
+    const texture = t.createTexture({ arrayLayerCount: ARRAY_LAYER_COUNT_2D });
+
+    const descriptor = { format, dimension, arrayLayerCount, mipLevelCount };
+
+    await t.expectValidationError(() => {
+      texture.createView(descriptor);
+    }, !success);
+  }
+).params([
+  { success: true },
+  { format: 'rgba8unorm', success: true },
+  { format: 'r8unorm', success: false },
+  { dimension: '2d-array', success: true },
+  { dimension: '2d', success: false },
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, success: false }, // setting array layers to non-0 means the dimensionality will default to 2D so by itself it causes an error.
+  { arrayLayerCount: ARRAY_LAYER_COUNT_2D, dimension: '2d-array', success: true },
+  {
+    arrayLayerCount: ARRAY_LAYER_COUNT_2D,
+    dimension: '2d-array',
+    mipLevelCount: MIP_LEVEL_COUNT,
+    success: true,
+  },
+]);
+
+g.test('default values validates the same as specifying values for only 1 array layer', async t => {
+  const { format, dimension, arrayLayerCount, mipLevelCount, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: 1 });
+
+  const descriptor = { format, dimension, arrayLayerCount, mipLevelCount };
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { success: true },
+  { format: 'rgba8unorm', success: true },
+  { format: 'r8unorm', success: false },
+  { dimension: '2d-array', success: true },
+  { dimension: '2d', success: true },
+  { arrayLayerCount: 0, success: true },
+  { arrayLayerCount: 1, success: true },
+  { arrayLayerCount: 2, success: false },
+  { mipLevelCount: MIP_LEVEL_COUNT, success: true },
+  { mipLevelCount: 1, success: true },
+]);
+
+g.test('creating cube map texture view', async t => {
+  const { dimension = '2d-array', arrayLayerCount, success } = t.params;
+
+  const texture = t.createTexture({ arrayLayerCount: 16 });
+
+  const descriptor = t.getDescriptor({
+    dimension,
+    arrayLayerCount,
+  });
+
+  await t.expectValidationError(() => {
+    texture.createView(descriptor);
+  }, !success);
+}).params([
+  { dimension: 'cube', arrayLayerCount: 6, success: true }, // it is OK to create a cube map texture view with arrayLayerCount == 6
```

Comment:
let's also make sure it fails when `arrayLayerCount` is not specified

Reply:
Done